examples/NonArduino/Pico/PicoHal.h

```diff
@@ -0,0 +1,142 @@
+#ifndef PICO_HAL_H
+#define PICO_HAL_H
+
+// include RadioLib
+#include <RadioLib.h>
+
+// include the necessary Pico libraries
+#include <pico/stdlib.h>
+#include "hardware/spi.h"
+#include "hardware/timer.h"
+
+// create a new Raspberry Pi Pico hardware abstraction 
+// layer using the Pico SDK
+// the HAL must inherit from the base RadioLibHal class
+// and implement all of its virtual methods
+class PicoHal : public RadioLibHal {
+public:
+  PicoHal(spi_inst_t *spiChannel, uint32_t misoPin, uint32_t mosiPin, uint32_t sckPin, uint32_t spiSpeed = 500 * 1000)
```

Comment:
The default SPI speed is quite low (500 kHz vs 2 MHz in Arduino). Is there a specific reason for that? Asking because of #938.